### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by AckonSamuel**
## Description
In this project,  Decorator design pattern was used to correct the names of people.

### General requirements

- [x] Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- [ ]Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- [ ]Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### Ruby requirements
- [x] Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### Project requirements
- [x] Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)
- [x] Think about how you can use two decorators in order to capitalize and trim people's names.
- [x] In the following steps you will get instructions for implementing the Decorator design pattern for this case.

#### Interface
- [x] Create a class `Nameable`.
   - Implement a method called `correct_name` that will raise a `NotImplementedError`.

#### Turn your Person class to Nameable
- [x] Make sure that your Person class inherits from Nameable
- [x] Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.

#### Prepare base Decorator
- [x] Make sure that it inherits from Nameable.
- [x] In the constructor assign a nameable object from params to an instance variable.
- [x] Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### Prepare CapitalizeDecorator and TrimmerDecorator
- [x] For the CapitalizeDecorator:
   - [x] Create a class that inherits from the base Decorator class.
    - [x] Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- [x] For the TrimmerDecorator:
    - [x] Create a class that inherits from the base Decorator class.
    - [x] Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: capdec.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
```
Comment (danmainah) [inline capdec.rb]: - [x] Kindly check on the functionality the  correct_name method is supposed to capitalize the first letter and not the entire name.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @AckonSamuel , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Github flow was correctly used


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: capdec.rb (hunk 1 above)
Comment (AckonSamuel) [reply capdec.rb]: thank you

**Review by mmsesay — APPROVED** [review]
Hi @AckonSamuel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
 - Good Readme :heavy_check_mark: 
 - No linter error :heavy_check_mark: 
 - Good PR title :heavy_check_mark: 
 - Good PR summary :heavy_check_mark:  


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**